pythonforandroid/recipes/numpy/__init__.py

```
@@ -37,54 +32,23 @@ def get_recipe_env(self, arch=None, with_flags_in_cc=True):
         # NPY_DISABLE_SVML=1 allows numpy to build for non-AVX512 CPUs
         # See: https://github.com/numpy/numpy/issues/21196
         env["NPY_DISABLE_SVML"] = "1"
-
+        env["TARGET_PYTHON_EXE"] = join(Recipe.get_recipe(
+                "python3", self.ctx).get_build_dir(arch.arch), "android-build", "python")
         return env
 
     def build_arch(self, arch):
-        self.hostpython_prerequisites = ["setuptools==59.2.0"]
-        self.install_hostpython_prerequisites()
+        # NumPy requires complex maths functions which were added in api 24
+        if self.ctx.ndk_api < 24:
```

Minor:

There's a simple way to fail faster? (Otherwise we're gonna implement it later)

---

You mean before download step?

---

That would be perfect